doc/quick-start/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM rust:1.67.1-slim-bullseye as builder
```

Comment:
There are two docker files right now. I guess we can replace the `script/Dockerfile` with this. Therefore, there is not point to keep a `doc/quick-start` folder.

Reply:
`scripts/Dockerfile` is required in CI to build the release image, if we want to replace it, we need to change the CI scripts steps.

Reply:
What's the main differences between these two files?

Reply:
One very troublesome issue is that in `validation_test.sh`, we need to compile `lock_client` on the host machine and mount it under `/mnt` in the container for validation. For some non-Linux host machines, we can only compile it within a Docker container and then copy it out. This also means that the `lock_client` needs to be included in the image, it is redundant, so this `doc/quick-start/Dockerfile` is not suitable for building and releasing images in CI.